docs/guides/developer/docs/installation/source-linux.md

```diff
@@ -0,0 +1,90 @@
+Developer installation guide
+===========================
+
+These instructions outline how to install Opencast on a Linux system.
+
+
+1.Get Opencast Source
+--------------------
+
+You can get the Opencast source code by cloning the Git
+repository.
+
+Cloning the Git repository:
+
+    git clone https://github.com/opencast/opencast.git
+    cd opencast
+    git checkout develop
+
+2.Install Dependencies
+--------------------
+
+Please make sure to install the following dependencies.
+
+Required:
+
+    java-1.8.0-openjdk-devel.x86_64 / openjdk-8-jdk
+    ffmpeg >= 3.2.4
+    maven >= 3.1
+    unzip
+    gcc-c++
+    tar
+    bzip2
+
+Required as a service for running Opencast:
+
+    ActiveMQ >= 5.10 (older versions untested)
+
+Required for some services. Some tests may be skipped and some features
+may not be usable if they are not installed. Hence, it's generally a good idea to
+install them.
+
+    tesseract >= 3
+    hunspell >= 1.2.8
+    sox >= 14.4
+    synfig
+
+### Dependency Download
+
+Pre-built versions of most dependencies that are not in the repositories can be downloaded from the respective project
+website:
+
+* [Get FFmpeg](http://ffmpeg.org/download.html)
+* [Get Apache Maven](https://maven.apache.org/download.cgi)
+* [Get Apache ActiveMQ](http://activemq.apache.org/download.html)
+
+3.ActiveMQ Configuration
+--------------------
+
+Opencast comes with a basic configuration for ActiveMQ. Please follow the first
+configuration step to copy the XML file. [Message Broker Configuration](https://docs.opencast.org/develop/admin/configuration/message-broker/).
+
+4.Build and Start Opencast
+--------------------
+
+Automatically build Opencast and how to start it.
+The -Pdev argument decreases the build time and skips the creation of multiple
+distribution tarballs by summarizing them into one.
+
+    cd opencast-dir
+    mvn clean install -Pdev
+    cd build/opencast-*
+    ./bin/start-opencast
+
+### Useful Commands for Building Opencast
+
+If you need a quick build of Opencast for test purposes, you can also use the
+following command to skip the Opencast building tests.
+
+    cd opencast-dir
+    mvn clean install -DskipTests=true
+
+To see the whole stacktrace of the installation you can use the following command
+to disable the trimming.
+
+    cd opencast-dir
+    mvn clean install -DtrimStackTrace=false
+
+5.Modify Code and Build Changes
+--------------------
+After you modified your code you can go back to step 4 to rebuild Opencast.
```

Comment:
Stupid development tricks from my .bash_aliases:

- Rebuild just the module you're working on with: `mvn clean install && mv -f target/*?basename $(pwd)?* ../../build/opencast-dist-*/system/org/opencastproject/*?basename $(pwd)?/*/` and then restart Opencast to skip the extremely costly bundle building step.  Note that the Markdown highlighting here eats backticks, so I've replaced them with question marks.
- If everything goes to hell and you need a completely fresh package from your Maven cache: `cd assemblies && mvn clean install -P dev && cd ..`

Admin UI development tricks: `https://docs.opencast.org/develop/developer/modules/admin-ui/development/`, again using a relative link.

Reply:
My general intention for this guide was to be a starting point for new developers which could *quickly* skim these docs (instead of reading through the admin install docs) to get things started. It's not meant to be comprehensive and I think we should avoid getting to complicated. I hate endless guides when I just try to get a new project up and running as a dev ;-P

That's why I don't think it's the right place to add something like this. This guide is meant as a quick start for developers and I wouldn't include special development instructions for a specific module and also not some command line chains despite them possibly being useful in some scenarios. Starting with Opencast development, that's likely more confusing than helpful and you would really need to explain what you are doing. That would be more something for a development tips and tricks section where you could also add things like Karaf's `bundle:watch` , cleaning a distribution, persisting data, …

But for this guide, a simple “rebuild Opencast” should suffice for all these special cases ;-)

Reply:
Fair enough.  If @dankaiser1808 is interested in writing up those docs then maybe that's space for another PR :)